image_flip/src/image_flip_node.cpp

```diff
@@ -0,0 +1,282 @@
+// Copyright (c) 2022, CHRISLab, Christopher Newport University
+// Copyright (c) 2014, JSK Lab.
+// Copyright (c) 2008, Willow Garage, Inc.
+// All rights reserved.
+//
+// Software License Agreement (BSD License 2.0)
+//
+// Redistribution and use in source and binary forms, with or without
+// modification, are permitted provided that the following conditions
+// are met:
+//
+//  * Redistributions of source code must retain the above copyright
+//    notice, this list of conditions and the following disclaimer.
+//  * Redistributions in binary form must reproduce the above
+//    copyright notice, this list of conditions and the following
+//    disclaimer in the documentation and/or other materials provided
+//    with the distribution.
+//  * Neither the name of the Willow Garage nor the names of its
+//    contributors may be used to endorse or promote products derived
+//    from this software without specific prior written permission.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+// "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+// LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+// FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+// COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+// INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+// BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+// LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+// CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+// LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+// ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+// POSSIBILITY OF SUCH DAMAGE.
+
+/********************************************************************
+* image_flip_node.cpp
+* this is a forked version of image_flip.
+* this image_flip_node supports:
+*  1) node
+*  2) tf and tf2
+*********************************************************************/
+
+#include "image_flip/image_flip_node.hpp"
+
+#include <vector>
+#include <string>
+#include <memory>
+#include <opencv2/core/core.hpp>
+
+namespace image_flip
+{
+
+ImageFlipNode::ImageFlipNode(rclcpp::NodeOptions options)
+: Node("ImageFlipNode", options)
+{
+  config_.output_frame_id = this->declare_parameter("output_frame_id", std::string(""));
+  config_.rotation_steps = this->declare_parameter("rotation_steps", 2);
+  config_.use_camera_info = this->declare_parameter("use_camera_info", true);
+  config_.in_image_topic_name =
+    this->declare_parameter("in_image_topic_name", std::string("image"));
+  config_.out_image_topic_name =
+    this->declare_parameter("out_image_topic_name", std::string("rotated_image"));
```

Comment:
IIRC, using a parameter was a hack around an issue with the rclcpp resolve_topic_name and how it dealt with images.  It will be a bit before I can dig into that again.  Do you happen to know if the resolve_topic_name issue was resolved?

Reply:
I think it's fixed now.  During testing I found I could not use the image_view, and rviz2 was crashing with rolling, so I added a parameter to let you select QoS settings for input and output topics.  Defaults to sensor_data (best effort) on input, and default (reliable) on output.